sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cra-integration/_tests_/cra-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,116 @@
+import { createMock, DeepMocked } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../test/helpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeCRAIncomeVerification,
+  saveFakeApplication,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import * as Client from "ssh2-sftp-client";
+import * as path from "path";
+import { CRAResponseIntegrationScheduler } from "../cra-response-integration.scheduler";
+import {
+  createFileFromStructuredRecords,
+  getStructuredRecords,
+  mockDownloadFiles,
+} from "@sims/test-utils/mocks";
+import { Job } from "bull";
+import { ApplicationStatus } from "@sims/sims-db";
+
+const CRA_FILENAME = "CRA_200_PBCSA00000.TXT";
+
+const padWithLeadingZeros = (num: number): string => {
+  // Pad the number with leading zeros to make it 9 digits long
+  return num.toString().padStart(9, "0");
+};
+
+describe(describeProcessorRootTest(QueueNames.CRAResponseIntegration), () => {
+  let app: INestApplication;
+  let processor: CRAResponseIntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let craResponseFolder: string;
+
+  beforeAll(async () => {
+    craResponseFolder = path.join(__dirname, "cra-receive-files");
+    process.env.CRA_RESPONSE_FOLDER = craResponseFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    processor = app.get(CRAResponseIntegrationScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it.only("should process CRA response file", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+
+    const application = await saveFakeApplication(
+      db.dataSource,
+      { student },
+      { applicationStatus: ApplicationStatus.InProgress },
+    );
+
+    // Create CRA income verifications for student.
+    const studentCRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application,
+      },
+      { initialValues: { dateReceived: null } },
+    );
+    await db.craIncomeVerification.save([studentCRAIncomeVerification]);
+    // Queued job.
+    const job = createMock<Job<void>>();
+    mockDownloadFiles(sftpClientMock, [CRA_FILENAME]);
+
+    mockDownloadFiles(sftpClientMock, [CRA_FILENAME], (fileContent: string) => {
+      const file = getStructuredRecords(fileContent);
+      const line2 = file.records[2]; // Get the 4th item (index 3) in the array
+
+      // Split the record on colon and change 9 digits after colon
+      const [beforeColon, afterColon] = line2.split(":");
+      const newVerificationId = padWithLeadingZeros(
+        studentCRAIncomeVerification.id,
+      );
```

Comment:
Can we use one replacement strategy similar to the one below?
https://github.com/bcgov/SIMS/blob/4e3b55ab77ffb49cf25f35eaa0fd0f8650e38e3f/sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/_tests_/ece-response-integration.scheduler.e2e-spec.ts#L136-L145

Reply:
To be clear, the comment was not targeting the removal of the ":" but the use of the replace to adjust the file content.

Reply:
The below is the expected code. Tested locally and it works.
```ts
mockDownloadFiles(sftpClientMock, [CRA_FILENAME], (fileContent: string) => {
      const file = getStructuredRecords(fileContent);
      file.records[2] = file.records[2].replace(
        "CRA_INCOME_VERIFICATION",
        studentCRAIncomeVerification.id.toString().padStart(9, "0"),
      );
      return createFileFromStructuredRecords(file);
    });
```